builder/ebs/builder.go

```diff
@@ -251,6 +251,7 @@ func (b *Builder) Run(ctx context.Context, ui packersdk.Ui, hook packersdk.Hook)
 			SpotTags:                          b.config.SpotTags,
 			Tags:                              b.config.RunTags,
 			SpotInstanceTypes:                 b.config.SpotInstanceTypes,
+			SpotAllocationStrategy:            b.config.RunConfig.SpotAllocationStrategy,
```

Comment:
I believe this setting should be replicated to other builders, assuming this is compatible with them.
Otherwise I would suggest either:

1. moving that argument to `builder/ebs/config.go` so it is not common
2. checking that it is not set in the configs for the builders that aren't compatible with it, this can be done in the `Prepare` function for each builder.

Reply:
Note: if the `ebs` builder's the only one to support spot instances, I would suggest going with approach 1